application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramMacroSolrMetadataExtractor.java

```diff
@@ -65,32 +73,81 @@ public class DiagramMacroSolrMetadataExtractor implements SolrEntityMetadataExtr
     @Inject
     private EntityNameValidationConfiguration entityNameValidationConfiguration;
 
+    @Inject
+    private Logger logger;
+
     @Override
     public boolean extract(XWikiDocument document, SolrInputDocument solrDocument)
```

Comment:
I realized you are missing unit tests for this class (+ what has been added in https://github.com/xwikisas/application-diagram/commit/1148b01f3ab5018a5c947cc39fc41ef83adf5414 )
Could you add in this PR unit tests for usecases specific to it, and make a separate PR for what other unit tests are missing?